src/inmanta/server/services/paramservice.py

```diff
@@ -200,54 +227,69 @@ async def _update_param(
 
         return recompile and value_updated
 
-    @handle(methods.set_param, param_id="id", env="tid")
+    @handle(methods.set_param, name="id", env="tid")
     async def set_param(
         self,
         env: data.Environment,
-        param_id: str,
+        name: str,
         source: ParameterSource,
         value: str,
-        resource_id: str,
+        resource_id: Optional[str],
         metadata: JsonType,
         recompile: bool,
+        expires: Optional[bool] = None,
     ) -> Apireturn:
-        result = await self._update_param(env, param_id, value, source, resource_id, metadata, recompile)
+        result = await self._update_param(env, name, value, source, resource_id, metadata, recompile, expires)
         warnings = None
         if result:
             compile_metadata = {
                 "message": "Recompile model because one or more parameters were updated",
                 "type": "param",
-                "params": [(param_id, resource_id)],
+                "params": [(name, resource_id)],
             }
             warnings = await self.server_slice._async_recompile(env, False, metadata=compile_metadata)
 
         if resource_id is None:
             resource_id = ""
 
-        params = await data.Parameter.get_list(environment=env.id, name=param_id, resource_id=resource_id)
+        params = await data.Parameter.get_list(environment=env.id, name=name, resource_id=resource_id)
 
         return attach_warnings(200, {"parameter": params[0]}, warnings)
 
     @handle(methods.set_parameters, env="tid")
     async def set_parameters(self, env: data.Environment, parameters: list[dict[str, Any]]) -> Apireturn:
         recompile = False
 
+        updating_facts: bool = False
```

Comment:
This method also needs to raise an exception is the provided expires parameter is invalid.

Reply:
I'm not confident about this part. I used a transaction so that we only insert the params iff they all are ok but I'm not sure if that's the way to go. Adding some tests for this would be nice but I don't quite see how to do this because of the `agent_server=True,` on the `set_parameters` endpoint